src/main/java/seedu/jarvis/model/financetracker/FinanceTracker.java

```diff
@@ -63,60 +53,37 @@ public void resetData(FinanceTracker financeTracker) {
         this.installmentList = new InstallmentList(financeTracker.getInstallmentList());
     }
 
-    /**
-     * Resets the existing data of this {@code AddressBook} with {@code newData}.
-     */
-    public void resetData(ReadOnlyAddressBook newData) {
-        requireNonNull(newData);
-
-        setPurchases(newData.getPurchaseList());
-        filteredPurchases.setPredicate(FinanceTrackerModel.PREDICATE_SHOW_ALL_PURCHASES);
-    }
+    //=========== Purchase List =======================================================================
 
-    //=========== Setter Methods ==================================================================================
+    // ========== Setter Methods ======================================================================
 
     public void setPurchaseList(PurchaseList purchaseList) {
         this.purchaseList = purchaseList;
     }
 
-    public void setInstallmentList(InstallmentList installmentList) {
-        this.installmentList = installmentList;
-    }
-
     /**
-     * Replaces the given person {@code target} in the list with {@code editedPerson}.
-     * {@code target} must exist in the address book.
-     * The person identity of {@code editedPerson} must not be the same as another existing person in the address book.
+     * Replaces the given purchase {@code target} in the list with {@code editedPurchase}.
+     * {@code target} must exist in the finance tracker.
+     * The identity of {@code editedPurchase} must not be the same as another existing purchase in the finance tracker.
      */
     public void setPurchase(Purchase target, Purchase editedPurchase) {
         requireNonNull(editedPurchase);
 
-        purchases.setPurchase(target, editedPurchase);
+        purchaseList.setPurchase(target, editedPurchase);
     }
 
-    //=========== Getter Methods ==================================================================================
+    //=========== Getter Methods ======================================================================
 
     public Purchase getPurchase(int paymentIndex) {
         return purchaseList.getPurchase(paymentIndex);
     }
 
-    public Installment getInstallment(int instalIndex) throws InstallmentNotFoundException {
-        return installmentList.getInstallment(instalIndex);
-    }
-
-    public double getMonthlyLimit() {
-        return monthlyLimit;
+    public ArrayList<Purchase> getPurchaseList() {
```

Comment:
Perhaps return a `List` instead? Not much issue here actually, but I think its good to return the interface instead of concrete class instead.

Reply:
ArrayLists used were changed to ObservableLists to accomodate ListFinancesCommand